model/event/host_event_agent_deploy.go

```diff
@@ -0,0 +1,95 @@
+package event
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// RecentHostAgentDeploys is a type used to capture the results of an
+type RecentHostAgentDeploys struct {
+	HostID       string `bson:"host_id" json:"host_id" yaml:"host_id"`
+	Last         string `bson:"last" json:"last" yaml:"last"`
+	Count        int    `bson:"count" json:"count" yaml:"count"`
+	Failed       int    `bson:"failed" json:"failed" yaml:"failed"`
+	Success      int    `bson:"success" json:"success" yaml:"success"`
+	Total        int    `bson:"total" json:"total" yaml:"total"`
+	message.Base `bson:"metadata" json:"metadata" yaml:"metadata"`
+}
+
+func GetRecentAgentDeployStatuses(hostId string, n int) (*RecentHostAgentDeploys, error) {
+	query := resourceTypeKeyIs(ResourceTypeHost)
+	query[TypeKey] = bson.M{"$in": []string{EventHostAgentDeployed, EventHostAgentDeployFailed}}
+	query[ResourceIdKey] = hostId
+
+	pipeline := []bson.M{
+		{"$match": query},
+		{"$sort": bson.M{TimestampKey: -1}},
+		{"$limit": n},
+		{"$group": bson.M{
+			"_id":    nil,
+			"count":  bson.M{"$sum": 1},
+			"states": bson.M{"$push": "$" + TypeKey},
+			"last":   bson.M{"$first": "$" + TypeKey},
+		}},
+		{"$addFields": bson.M{
+			"failed": bson.M{"$size": bson.M{
+				"$filter": bson.M{
+					"input": "$states",
+					"cond":  bson.M{"$eq": []string{"$$this", EventHostAgentDeployFailed}},
+				},
+			}},
+			"success": bson.M{"$size": bson.M{
+				"$filter": bson.M{
+					"input": "$states",
+					"cond":  bson.M{"$eq": []string{"$$this", EventHostAgentDeployed}},
```

Comment:
Make sure the test fails in the inverse case.

Reply:
I believe that the current test is correct, and I had fixed this after pushing but the PR was in a weird state.